File: Test/AdventureWorksFunctionalModel/MockPrincipalProvider.cs
using System.Security.Principal;
using NakedFunctions;

namespace AW {
    public class MockPrincipalProvider : IPrincipalProvider {
        public IPrincipal CurrentUser =>
            new GenericPrincipal(new GenericIdentity(@"adventure-works\ken0"), new[] { "root" });
    }
}

File: Test/NOF2.Demo.Model/MockPrincipalProvider.vb
Imports System.Security.Principal


Namespace AW
	Public Class MockPrincipalProvider
		Implements IPrincipalProvider

		Public ReadOnly Property CurrentUser() As IPrincipal
			Get
				Return New GenericPrincipal(New GenericIdentity("adventure-works\ken0"), { "root" })
			End Get
		End Property
	End Class
End Namespace